Bottom: 740e44fd3af78d476f7d3f19a4cfc7731a91d7f8
Top:    42f863c74b834de203cf9eca736df8f73135ad6b
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-02-23 22:46:45 +1300

Refresh of double-streamresult

---

diff --git a/testtools/testresult/doubles.py b/testtools/testresult/doubles.py
index ede3c18..d86f7fa 100644
--- a/testtools/testresult/doubles.py
+++ b/testtools/testresult/doubles.py
@@ -166,12 +166,9 @@ class StreamResult(object):
     def stopTestRun(self):
         self._events.append(('stopTestRun',))
 
-    def file(self, file_name, file_bytes, eof=False, mime_type=None,
-        test_id=None, route_code=None, timestamp=None):
-        self._events.append(('file', file_name, file_bytes, eof, mime_type,
-            test_id, route_code, timestamp))
-
-    def status(self, test_id, test_status, test_tags=None, runnable=True,
-        route_code=None, timestamp=None):
+    def status(self, test_id=None, test_status=None, test_tags=None,
+        runnable=True, file_name=None, file_bytes=None, eof=False,
+        mime_type=None, route_code=None, timestamp=None):
         self._events.append(('status', test_id, test_status, test_tags,
-            runnable, route_code, timestamp))
+            runnable, file_name, file_bytes, eof, mime_type, route_code,
+            timestamp))
diff --git a/testtools/tests/test_testresult.py b/testtools/tests/test_testresult.py
index 286b254..ef65d04 100644
--- a/testtools/tests/test_testresult.py
+++ b/testtools/tests/test_testresult.py
@@ -540,11 +540,11 @@ class TestDoubleStreamResultEvents(TestCase):
         result = LoggingStreamResult()
         result.startTestRun()
         now = datetime.datetime.now(utc)
-        result.file("foo", "bar", eof=True, mime_type="text/json",
+        result.status(file_name="foo", file_bytes="bar", eof=True, mime_type="text/json",
             test_id="id", route_code='abc', timestamp=now)
         self.assertEqual(
             [('startTestRun',),
-             ('file', 'foo', 'bar', True, 'text/json', 'id', 'abc', now)],
+             ('status', 'id', None, None, True, 'foo', 'bar', True, 'text/json', 'abc', now)],
             result._events)
 
     def test_status(self):
@@ -555,7 +555,7 @@ class TestDoubleStreamResultEvents(TestCase):
             runnable=False, route_code='abc', timestamp=now)
         self.assertEqual(
             [('startTestRun',),
-             ('status', 'foo', 'success', set(['tag']), False, 'abc', now)],
+             ('status', 'foo', 'success', set(['tag']), False, None, None, False, None, 'abc', now)],
             result._events)
